app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/yeartodatelliability/YearToDateLiabilityController.scala

```diff
@@ -1779,9 +1780,17 @@ class YearToDateLiabilityController @Inject() (
                 )
                   draftReturn
                 else {
-                  val newAnswers = nonCalculatedAnswers
-                    .unset(_.yearToDateLiability)
-                    .copy(yearToDateLiability = Some(yearToDateLiability))
+                  val newAnswers =
```

Comment:
we'll always be in a further return in this endpoint so you don't need to check over here - it's already being checked above.  It should always be this:
```scala
 nonCalculatedAnswers
                        .unset(_.yearToDateLiability)
                        .unset(_.mandatoryEvidence)
                        .unset(_.taxDue)
                        .unset(_.checkForRepayment)
                        .copy(yearToDateLiability = Some(yearToDateLiability))
```

Reply:
the following endpoints need to be updated as well:
```
taxableGainorLossSubmit() (Net gain or net loss)
hasEstimatedDetailsSubmit() 
nonCalculatedEnterTaxDueSubmit() (Confirm tax owed in this return)
``` 
where you do the updates in those endpoints, there might already be checks that it's a further return around them so that's something to look out for